eeg_flow/tasks/epochs_evoked.py

```diff
@@ -0,0 +1,434 @@
+    )
+
+    fname_rt_plot = derivatives_subfolder / "plots" / (fname_stem + "_step9_RT.svg")
+    ax_rt.figure.suptitle(fname_stem, fontsize=16, y=1)
+    ax_rt.figure.savefig(fname_rt_plot, transparent=True)
+    ax_rt
+
+def get_indiv_behav(metadata, num_Hits, num_CorrectRejections, num_Misses, num_FalseAlarms, 
+                    fname_stem, derivatives_subfolder, response_mean, response_std):
+    correct_response_count = metadata["response_correct"].sum()
+
+    print(
+        f"\nCorrect responses: {correct_response_count}\n"
+        f"Incorrect responses: {len(metadata) - correct_response_count}\n"
+    )
+
+    print("Hits, Misses, Correct Rejections, False Alarms")
+    print(num_Hits, num_Misses, num_CorrectRejections, num_FalseAlarms,"\n")
+    SDT(num_Hits, num_Misses, num_FalseAlarms, num_CorrectRejections)
+    metadata.groupby(by="event_name").count()
+    
+    # write behav file
+    FNAME_BEHAV = derivatives_subfolder / (fname_stem + "_step9_b-behav.txt")
+
+    file_behav = open(FNAME_BEHAV, "w")
+
+    file_behav.write("Hits, Misses, Correct Rejections, False Alarms\n")
+    file_behav.write(f"{str(num_Hits)}\t{str(num_Misses)}\t{str(num_CorrectRejections)}\t{str(num_FalseAlarms)}")
+
+
+    file_behav.write("\n\nStandard, Novel, Target\n")
+    metadata_count_correct = metadata.groupby(by="event_name").count()[
+        "response_correct"
+    ]
+    count_corr_standard = str(metadata_count_correct["standard"])
+    count_corr_target = str(metadata_count_correct["target"])
+    count_corr_novel = str(metadata_count_correct["novel"])
+
+    file_behav.write(f"{count_corr_standard}\t{count_corr_novel}\t{count_corr_target}")
+
+    file_behav.write("\n\nResponse_mean, Response_std\n")
+    file_behav.write(str(response_mean) + "\t" + str(response_std))
+
+    file_behav.write("\n\nd'\n")
+    file_behav.write(
+        str(SDT(num_Hits, num_Misses, num_FalseAlarms, num_CorrectRejections)["d"])
+    )
+
+    file_behav.close()  # to change file access modes
+
+
+def epoching(raw, events, event_id, metadata):
+    """Epoching.
+    Parameters
+    ----------
+    %(raw)s
+    %(events)s
+    %(event_id)s
+    %(metadata)s
+    """
+    epochs_tmin, epochs_tmax = -0.2, 0.8
+    epochs = Epochs(
+        raw=raw,
+        tmin=epochs_tmin,
+        tmax=epochs_tmax,
+        events=events,
+        event_id=event_id,
+        metadata=metadata,
+        reject=None,
+        preload=True,
+        baseline=(None, 0),
+        picks="eeg",
+    )
+    return epochs
+
+def get_rejection(epochs):
+    starttime = time.time()
+    reject = get_rejection_threshold(epochs, decim=1, ch_types="eeg", random_state=888)
+    endtime = time.time()
+    diff = endtime - starttime
+    minutes = str(int(diff // 60)).zfill(2)
+    seconds = str(int(diff % 60)).zfill(2)
+    print("\nPeak-to-peak rejection threshold computed: %s", reject)
+    print(f"Elapsed {minutes}:{seconds}\n")
```

Comment:
I'm seeing this kind of code so often in your scripts, that you could write a simple context manager which:
- get the current time on entry
- get the current time on exit
- log the elapsed time on exit

Something along the line of:

```
from contextlib import contextmanager

@contextmanager
def log_execution_time():
    start = time.time()
    yield
    stop = time.time()
    elapsed = stop  - start
    ...
    logger.info(...)
```

Note that syntax defining the context manager as a function with the decorator `
@contextmanager`, it's equivalent to the class syntax:

```
class LogExecutionTime():

    def __enter__(self):
        self.start = time.time()

    def __exit__(self, exc_type, exc_value, traceback):
        stop = time.time()
        elapsed = stop - self.start()
        ...
        logger.info(...)
```

This context-manager could live in a `utils` (utilities) module, and could be use like so:

```
with log_execution_time():
    foo()

with LogExecutionTime():
    foo()
```